bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/listener/OffDelayUpdateStateListener.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mqtt.homeassistant.internal.listener;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.mqtt.generic.ChannelStateUpdateListener;
+import org.openhab.binding.mqtt.generic.values.Value;
+import org.openhab.binding.mqtt.homeassistant.generic.internal.MqttBindingConstants;
+
+/**
+ * A listener to set the binary sensor value to 'off' after a timeout.
+ *
+ * @author Jochen Klein - Initial contribution
+ */
+@NonNullByDefault
+public class OffDelayUpdateStateListener extends ChannelStateUpdateListenerProxy {
+
+    private final int offDelay;
+    private final Value value;
+    private AtomicReference<@Nullable ScheduledFuture<?>> delay = new AtomicReference<>();
+    private final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(MqttBindingConstants.HOMEASSISTANT_THREADPOOL_NAME);
+
+    public OffDelayUpdateStateListener(ChannelStateUpdateListener original, int offDelay, Value value) {
+        super(original);
+        this.offDelay = offDelay;
+        this.value = value;
+    }
+
+    @Override
+    public void updateChannelState(final ChannelUID channelUID, State state) {
+        super.updateChannelState(channelUID, state);
```

Comment:
How come in this method you call the super first while the other one you call it last?

Reply:
I think, I had a difference during development. But I cannot remember any more.
It's changed.